### ISSUE TITLE
App disappears from list if URLCheck opens that app, and that app opens URLCheck

### ISSUE THREAD

**Issue by pressRtowin** [TrianguloY/UrlChecker#335]
### Steps to reproduce

1. Not related, but to ensure it's not a factor, disable "Hide the source app (referrer)".
2. Install a different link handling app alongside URLCheck.
   - Better Open With, Open Link With, or LinkSheet Nightly* (see below)
3. Open a link in URLCheck.
4. Select the other link handler installed previously.
5. From that link handler, select URLCheck again.
6. The other link handler is now missing from URLCheck

\* For some reason, in the case of LinkSheet Nightly, steps 4 and 5 must be repeated again before it disappears from URLCheck. Therefore the steps to reproduce the bug with LinkSheet Nightly are:

1. Open a link in URLCheck.
2. Select LinkSheet Nightly.
3. Select URLCheck.
4. Select LinkSheet Nightly.
5. Select URLCheck.
6. LinkSheet Nightly is now missing in URLCheck.

### Expected behavior

Other link handlers should remain in URLCheck's list when switching between them.

### Actual behavior

Apps disappear if switching to and from URLCheck in the manner described above.

### URLCheck version

2.14

### Android version

Android 14

### Android Custom/Specific ROM or Device

One UI 6.0

### Other details

I don't know how I keep finding this stuff. 😅

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by TrianguloY** [comment]
This is a know "issue" that is already fixed with https://github.com/TrianguloY/UrlChecker/commit/8b8ad9d804c2344af56b6e9c7fbf2b9793c290d1 (but I'll keep this open just in case until the version is released).

All started with some apps "rejecting" urls, for example Github. The Github app is configured to open all Github links, but the app cannot handle some of them (like wikis) so it opens the default browser instead (well, it did, I think it now asks you...but with a predefined set of apps? It's strange). Anyway, this meant that, if you open an unsupported link from URLCheck to Github, it will open URLCheck again (immediately without any visual indication) and from the user point of view it was like the open button didn't work.

Originally I disabled the referrer, but that didn't work. Later I added custom logic so that if URLCheck opens a link in an app, and that app reopens the same link in URLCheck in less that 5 seconds, that app is removed.
This is, almost without a doubt, what happens here (very easy to check: after step 6 wait at least 5 second and trigger any change, for example edit the url).

This logic however is flawed and it fails indeed some common circumstances. Next version fixes this but also allows to disable it altogether if needed.
